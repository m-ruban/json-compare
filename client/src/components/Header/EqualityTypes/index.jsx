import React from 'react';
import PropTypes from 'prop-types';
import { FormGroup, Input, Form } from 'reactstrap';

import withConsumer from 'hoc/withConsumer';
import { EQUALITY_TYPE_STRICT, EQUALITY_TYPE_NOT_STRICT, EQUALITY_TYPE_IGNORE } from 'constants/equality';
import 'components/Header/EqualityTypes/EqualityTypes.less';

const EqualityTypes = ({ equalityTypes, changeEqualityTypes }) => (
    <Form className="settings" inline>
        <FormGroup className="mb-2 mr-sm-2 mb-sm-0">
            <Input
                type="select"
                className="settings__types"
                bsSize="sm"
                value={equalityTypes}
                onChange={changeEqualityTypes}
            >
                <option value={EQUALITY_TYPE_STRICT}>{EQUALITY_TYPE_STRICT}</option>
                <option value={EQUALITY_TYPE_NOT_STRICT}>{EQUALITY_TYPE_NOT_STRICT}</option>
                <option value={EQUALITY_TYPE_IGNORE}>{EQUALITY_TYPE_IGNORE}</option>
            </Input>
        </FormGroup>
    </Form>
);

EqualityTypes.propTypes = {
    equalityTypes: PropTypes.string,
    changeEqualityTypes: PropTypes.func,
};

export default withConsumer(EqualityTypes);
