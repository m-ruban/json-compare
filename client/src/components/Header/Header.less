@import "~styles/variables.less";

.header {
  flex-grow: 0;
  border-bottom: 1px solid @border-color;
  color: @secondary-color;

  &__app {
    color: @secondary-color;
  }

  &__nav-bar {
    background-color: @secondary-background-color;
  }

  &__logo {
    width: 20px;
    position: relative;
    top: -2px;
    margin-left: 1px;
  }
}
