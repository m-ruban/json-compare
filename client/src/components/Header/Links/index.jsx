import React, { useCallback, useState, Fragment } from 'react';
import PropTypes from 'prop-types';
import { Nav, NavItem, Collapse, NavLink, NavbarToggler } from 'reactstrap';

import withConsumer from 'hoc/withConsumer';
import trl from 'modules/translation';
import Examples from 'components/Header/Links/Examples';
import FAQ from 'components/Header/Links/FAQ';
import 'components/Header/Links/Links.less';

const Links = ({ feedback }) => {
    const [isOpen, setOpen] = useState(false);
    const toggle = useCallback(() => {
        setOpen(!isOpen);
    }, [isOpen]);

    return (
        <Fragment>
            <NavbarToggler onClick={toggle} />
            <Collapse isOpen={isOpen} navbar>
                <Nav className="ml-auto" navbar>
                    <Examples />
                    <NavItem>
                        <NavLink>
                            <FAQ />
                        </NavLink>
                    </NavItem>
                    <NavItem>
                        <NavLink className="feedback" href={feedback} target="_blank">
                            {trl('Header.Links.feedback')}
                        </NavLink>
                    </NavItem>
                </Nav>
            </Collapse>
        </Fragment>
    );
};

Links.propTypes = {
    feedback: PropTypes.string,
};

export default withConsumer(Links);
