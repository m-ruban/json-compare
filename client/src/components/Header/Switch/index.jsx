import React, { useCallback, Fragment } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import withConsumer from 'hoc/withConsumer';
import trl from 'modules/translation';

const Switch = ({ isCompare, toggleCompare }) => {
    const onParseClick = useCallback(() => {
        if (!isCompare) {
            return;
        }
        toggleCompare();
    }, [isCompare, toggleCompare]);
    const onCompareClick = useCallback(() => {
        if (isCompare) {
            return;
        }
        toggleCompare();
    }, [isCompare, toggleCompare]);
    return (
        <Fragment>
            <span className={classNames('parse', { pointer: isCompare })} onClick={onParseClick}>
                {trl('Header.Switch.parse')}
            </span>
            <span className="divider">&</span>
            <span className={classNames('compare', { pointer: !isCompare })} onClick={onCompareClick}>
                {trl('Header.Switch.compare')}
            </span>
        </Fragment>
    );
};

Switch.propTypes = {
    isCompare: PropTypes.bool,
    toggleCompare: PropTypes.func,
};

export default withConsumer(Switch);
