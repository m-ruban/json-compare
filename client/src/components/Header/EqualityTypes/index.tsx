import * as React from 'react';
import { Form, FormGroup, Input } from 'reactstrap';

import 'components/Header/EqualityTypes/EqualityTypes.less';
import { EQUALITY_TYPE_IGNORE, EQUALITY_TYPE_NOT_STRICT, EQUALITY_TYPE_STRICT } from 'constants/equality';
import withConsumer from 'hoc/withConsumer';

interface IEqualityTypesProps {
    equalityTypes: string;
    changeEqualityTypes: (event: React.SyntheticEvent) => void;
}

const EqualityTypes = ({ equalityTypes, changeEqualityTypes }: IEqualityTypesProps) => (
    <Form className="settings" inline>
        <FormGroup className="mb-2 mr-sm-2 mb-sm-0">
            <Input
                type="select"
                className="settings__types"
                bsSize="sm"
                value={equalityTypes}
                onChange={changeEqualityTypes}
            >
                <option value={EQUALITY_TYPE_STRICT}>{EQUALITY_TYPE_STRICT}</option>
                <option value={EQUALITY_TYPE_NOT_STRICT}>{EQUALITY_TYPE_NOT_STRICT}</option>
                <option value={EQUALITY_TYPE_IGNORE}>{EQUALITY_TYPE_IGNORE}</option>
            </Input>
        </FormGroup>
    </Form>
);

export default withConsumer(EqualityTypes);
