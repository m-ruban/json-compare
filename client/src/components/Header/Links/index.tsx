import React from 'react';
import { Collapse, Nav, NavbarToggler, NavItem, NavLink } from 'reactstrap';

import Examples from 'components/Header/Links/Examples';
import FAQ from 'components/Header/Links/FAQ';
import withConsumer from 'hoc/withConsumer';
import trl from 'modules/translation';

import 'components/Header/Links/Links.less';

interface ILinksProps {
    feedback: string;
}

const Links = ({ feedback }: ILinksProps) => {
    const [isOpen, setOpen] = React.useState(false);
    const toggle = React.useCallback(() => {
        setOpen(!isOpen);
    }, [isOpen]);

    return (
        <>
            <NavbarToggler onClick={toggle} />
            <Collapse isOpen={isOpen} navbar>
                <Nav className="ml-auto" navbar>
                    <Examples />
                    <NavItem>
                        <NavLink>
                            <FAQ />
                        </NavLink>
                    </NavItem>
                    <NavItem>
                        <NavLink className="feedback" href={feedback} target="_blank">
                            {trl('Header.Links.feedback')}
                        </NavLink>
                    </NavItem>
                </Nav>
            </Collapse>
        </>
    );
};

export default withConsumer(Links);
