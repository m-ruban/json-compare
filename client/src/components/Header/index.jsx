import React from 'react';
import { Link } from 'react-router-dom';
import { Navbar } from 'reactstrap';

import { AppConsumer } from 'context';
import EqualityTypes from 'components/Header/EqualityTypes';
import Switch from 'components/Header/Switch';
import Links from 'components/Header/Links';
import Logo from 'components/Header/icon.png';

const Header = () => (
    <header>
        <AppConsumer>
            {({ isCompare, toggleCompare, feedback }) => (
                <Navbar color="light" expand="md" light>
                    <Link className="navbar-brand" to="/">
                        JS
                        <img src={Logo} alt="icon" />N
                    </Link>
                    <Switch isCompare={isCompare} toggleCompare={toggleCompare} />
                    {isCompare && <EqualityTypes />}
                    <Links feedback={feedback} />
                </Navbar>
            )}
        </AppConsumer>
    </header>
);

export default Header;
