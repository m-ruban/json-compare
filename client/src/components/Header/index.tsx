import React from 'react';
import { Navbar, NavbarBrand } from 'reactstrap';

import EqualityTypes from 'components/Header/EqualityTypes';
import 'components/Header/Header.less';
import Logo from 'components/Header/Icon.png';
import Links from 'components/Header/Links';
import Switch from 'components/Header/Switch';
import withConsumer from 'hoc/withConsumer';

interface IHeaderProps {
    isCompare: string;
}

const Header = ({ isCompare }: IHeaderProps) => (
    <header className="header">
        <Navbar className="header__nav-bar" expand="md" light>
            <NavbarBrand>
                JS
                <img className="header__logo" src={Logo} alt="icon" />N
            </NavbarBrand>
            <Switch />
            {isCompare && <EqualityTypes />}
            <Links />
        </Navbar>
    </header>
);

export default withConsumer(Header);
