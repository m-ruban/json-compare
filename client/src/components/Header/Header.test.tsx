import React from 'react';

import { render } from 'enzyme';

import Header from 'components/Header';
import { Provider } from 'store/Context';
import compare from 'tests/mocks/compare';

describe('<Header />', () => {
    it('Renders correctly', () => {
        const output = render(
            <Provider value={compare}>
                <Header />
            </Provider>,
        );
        expect(output).toMatchSnapshot();
    });
});
