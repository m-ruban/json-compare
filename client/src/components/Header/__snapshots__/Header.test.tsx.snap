// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Renders correctly 1`] = `
<header
  class="header"
>
  <nav
    class="header__nav-bar navbar navbar-expand-md navbar-light"
  >
    <a
      class="navbar-brand"
    >
      <div
        class="header__app"
      >
        JS
        <img
          alt="icon"
          class="header__logo"
          src="test-file-stub"
        />
        N
      </div>
    </a>
    <span
      class="switch switch_parse switch_pointer"
    >
      parse
    </span>
    <span
      class="switch__divider"
    >
      &
    </span>
    <span
      class="switch switch_compare"
    >
      compare
    </span>
    <form
      class="settings form-inline"
    >
      <div
        class="mb-2 mr-sm-2 mb-sm-0 form-group"
      >
        <select
          class="settings__types form-control-sm form-control"
        >
          <option
            value="STRICT"
          >
            STRICT
          </option>
          <option
            value="NOT_STRICT"
          >
            NOT_STRICT
          </option>
          <option
            value="IGNORE"
          >
            IGNORE
          </option>
        </select>
      </div>
    </form>
    <button
      class="navbar-toggler"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="collapse navbar-collapse"
    >
      <ul
        class="ml-auto navbar-nav"
      >
        <li
          class="examples dropdown nav-item"
        >
          <a
            aria-expanded="false"
            aria-haspopup="true"
            class="examples__toggle dropdown-toggle nav-link"
            href="#"
          >
            Examples
          </a>
          <div
            aria-hidden="true"
            class="bg-dark examples__dropdown dropdown-menu"
            role="menu"
            tabindex="-1"
          >
            <button
              class="examples__dropdown-item dropdown-item"
              role="menuitem"
              tabindex="0"
              type="button"
            >
              Tool Example
            </button>
            <button
              class="examples__dropdown-item dropdown-item"
              role="menuitem"
              tabindex="0"
              type="button"
            >
              Object Example
            </button>
          </div>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
          >
            <span
              class="faq-link"
            >
              FAQ
            </span>
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="feedback nav-link"
            target="_blank"
          >
            Feedback
          </a>
        </li>
      </ul>
    </div>
  </nav>
</header>
`;
